sdk/iot/core/inc/az_iot_core.h

```diff
@@ -58,6 +58,19 @@ typedef enum
   AZ_IOT_STATUS_TIMEOUT = 504,
 } az_iot_status;
 
+/**
+ * @brief Get the #az_iot_status from an int.
+ *
+ * @param[in] status_int The int with the status number.
+ * @param[out] status The #az_iot_status* with the status enum.
+ * @return The #az_result with the result of the get operation.
+ *  @retval #AZ_OK If the int is an #az_iot_status enum. `status` will be set to the according
+ * enum.
+ *  @retval #AZ_ERROR_ITEM_NOT_FOUND If the int is NOT an #az_iot_status enum.
+ */
+
+AZ_NODISCARD az_result az_iot_get_status_from_uint32(uint32_t status_int, az_iot_status* status);
```

Comment:
I don't think this should be part of our public API.
The easiest way is to simply cast from an int/uint32_t. 

1. Why would we want the extra code on the device?
2. This is hiding the actual error if it's not in the list - I don't think that's a good idea.